designs/full-cluster-lifecycle-api-cloudstack.md

```diff
@@ -0,0 +1,81 @@
+# Cluster Lifecycle API CloudStack
+
+## Introduction
+
+### Problem
+
+Currently the only way to manage EKS Anywhere CloudStack cluster is by using CLI. We need to support native K8s experience on CloudStack cluster lifecycle.
+
+### Goals and Objectives
+
+* Create/Upgrade/Delete CloudStack workload clusters using: kubectl apply -f cluster.yaml
+* Create/Update/Upgrade/Delete CloudStack workload clusters using GitOps/Terraform
+
+#### Not in Scope
+
+* Create/Upgrade/Delete self-managed clusters with API
+
+#### Limitation
+
+* We currently only supports k8s version < 1.24 on CloudStack
+* We only support Redhat, not on Ubuntu or Bottlerocket yet.
```

Comment:
Similar thing, did this play a part in the design of this feature?

Reply:
This is not part of this design, but list our current limitation.

Reply:
I see. But was it a limitation for this design?
Would anything in the design change if this limitations wasn't present?
Once we remove the limitation, do we need to change anything?

Reply:
will remove it from this design as it's more of machine requirement, which is not part of this design